Simplenote/src/androidTest/java/com/automattic/simplenote/integration/AddTagActivityTest.kt

```diff
@@ -0,0 +1,136 @@
+package com.automattic.simplenote.integration
+
+import android.app.Activity
+import androidx.test.core.app.ActivityScenario
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.action.ViewActions.replaceText
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.contrib.ActivityResultMatchers.hasResultCode
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.filters.MediumTest
+import androidx.test.runner.AndroidJUnit4
+import com.automattic.simplenote.AddTagActivity
+import com.automattic.simplenote.BaseUITest
+import com.automattic.simplenote.R
+import com.automattic.simplenote.utils.hasTextInputLayoutErrorText
+import org.hamcrest.CoreMatchers.not
+import org.junit.Assert
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@MediumTest
+class AddActivityTest : BaseUITest() {
```

Comment:
1) Praise (❤️): I really REALLY love how you translated the test scenarios in the accompanying #1363 issue to tests within this test suite, kudos! I already feel MUCH more comfortable in having the associated `AddTagActivity` be refactored and modernized in whichever way we feel like!
2) Warning (⚠️): Consider renaming the `AddActivityTest` to `AddTagActivityTest`. Otherwise, the IDE will not be able to associate this test suite with its main `AddTagActivity` class. Also, it seems that the file name is indeed `AddTagActivityTest` already.
3) Warning (⚠️): This `import androidx.test.runner.AndroidJUnit4` is deprecated. Consider fixing this deprecation.
4) Question (❓): Have you even considered the `given/when/then` pattern? For example, instead of `addNewValidTag` one could rename that to something more descriptive like `given new valid tag, when the save button is clicked, then the tag is saved`.
5) Question (❓): How can we be sure about our code coverage here? What means did you take to make sure that we haven't missed a case?
6) Suggestion (💡): In most cases the tests described within the accompanying #1363 issue and within this test suite are to 100% consistent, but on some rare cases there are not. Consider making those consistent as they might come handy later on. For example, see `Test 4` and `addTagWithSpace()`.

Reply:
> 2. Consider renaming the AddActivityTest to AddTagActivityTest.

Yes this was a typo. I renamed the class. 

> 3. This import androidx.test.runner.AndroidJUnit4 is deprecated.

Fixed. Added `AndroidJUnit4ClassRunner`

Reply:
> Have you even considered the given/when/then pattern?

Here I follow the name convention of the other tests since I was not sure to introduce a new name convention for tests. Do you think we should introduce a name convention for tests?

> In most cases the tests described within the accompanying #1363 issue
Fixed in ce39f707 and d66bcd4433250d4715706473b1a65fde825e9880.

Reply:
> How can we be sure about our code coverage here? What means did you take to make sure that we haven't missed a case?

I am not measuring coverage yet. What I did to generate the test cases was manual testing and reading the code to find all possible branches of execution. I think that might work for the first two screens since they are fairly simple, but we I could try to measure it for the `TagsActivity` tests since it is more complex.

Reply:
> Here I follow the name convention of the other tests since I was not sure to introduce a new name convention for tests. Do you think we should introduce a name convention for tests?

We started using the `given/when/then` convention in the other repos and if you feel that this is a good convention, I would be totally okay with you introducing it for this repo as well. 😃

Reply:
> Fixed in ce39f70 and d66bcd4.

🙏

Reply:
> I am not measuring coverage yet. What I did to generate the test cases was manual testing and reading the code to find all possible branches of execution. I think that might work for the first two screens since they are fairly simple, but we I could try to measure it for the TagsActivity tests since it is more complex.

Great, thanks for the response and for being mindful about this! 🌟